src/app/shared/parser.ts

```diff
@@ -0,0 +1,40 @@
+import { Settings } from '../models/settings';
+import { Compound } from '../models/compound';
+
+export class Parser {
+
+  static parseURL(settings: Settings, classPath: string) {
+    return settings.serverURL + '?' + settings.queryStr + '&classPath=luciusapi.' + classPath;
+  }
+
+  static parseClassPath(url: string): string {
+    let queryString = url.substring(url.indexOf('?') + 1 );
+    let start = queryString.indexOf('.') + 1;
+    let end = queryString.length;
+    return queryString.substring(start, end);
+  }
+
+  static parseRelatedCompounds(relatedCompounds: Compound): Array<string> {
+    let relatedCompoundsArray = [];
+    for (let entry of relatedCompounds.result) {
+      relatedCompoundsArray.push(entry[0].toString());
```

Comment:
`push` is less performant than using the `index` and initiating the array:
https://jsperf.com/array-push-vs-index-for-large-arrays/3